x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypes.java

```diff
@@ -51,6 +51,10 @@ public final class EsqlDataTypes {
     public static final DataType GEO_SHAPE = new DataType("geo_shape", Integer.MAX_VALUE, false, false, true);
     public static final DataType CARTESIAN_SHAPE = new DataType("cartesian_shape", Integer.MAX_VALUE, false, false, true);
 
+    public static final DataType COUNTER_LONG = new DataType("counter_long", LONG.size(), false, false, true);
+    public static final DataType COUNTER_INTEGER = new DataType("counter_integer", INTEGER.size(), false, false, true);
+    public static final DataType COUNTER_DOUBLE = new DataType("counter_double", DOUBLE.size(), false, false, true);
```

Comment:
Probably worth a javadoc for folks who haven't been following along.

Reply:
++, added in https://github.com/elastic/elasticsearch/pull/107877/commits/472fbd2de7b86babfbf5076395b0519ac7687595